trin-core/src/jsonrpc/eth.rs

```diff
@@ -43,3 +46,31 @@ pub async fn get_block_by_hash(
         Err(err) => Err(err),
     }
 }
+
+/// eth_getBlockByNumber
+pub async fn get_block_by_number(
+    params: Params,
+    header_oracle: Arc<RwLock<HeaderOracle>>,
+) -> anyhow::Result<Value> {
+    let params: GetBlockByNumberParams = params.try_into()?;
+    if params.block_number > MERGE_BLOCK_NUMBER {
+        return Err(anyhow!(
+            "eth_getBlockByNumber not currently supported for blocks after the merge (#{:?})",
+            MERGE_BLOCK_NUMBER
+        ));
+    }
+    let block_hash = header_oracle
+        .read()
+        .await
+        .get_hash_at_height(params.block_number)
+        .await?;
+    let params = Params::Array(vec![
+        Value::String(format!("{:?}", block_hash)),
+        Value::Bool(false),
+    ]);
+    get_block_by_hash(
```

Comment:
I guess we are currently not returning transactions and this makes the transaction details flag not have any effect on the result from this endpoint?

Reply:
Yup, but good catch that this would lead to unintended behavior for users over jsonrpc. So, I added a check to return an error if users request full transactions, to prevent this behavior for now. It can easily be removed down the road when we support full transactions, but I don't consider implementing that support a priority atm.